mobile/src/full/java/org/openhab/habdroid/core/GcmListenerService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2010-2018, openHAB.org and others.
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.habdroid.core;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.content.ContextCompat;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.ui.OpenHABMainActivity;
+import org.openhab.habdroid.util.Constants;
+
+public class GcmListenerService extends com.google.android.gms.gcm.GcmListenerService {
```

Comment:
Does this need to be the same class name? I kind of dislike that, as it doesn't make clear from the class name, why we extend it on our side. I also do not really like to prefix a class with the app name, but here it would maybe make sense: OpenHABGcmListenerService?

Reply:
`GcmMessageListenerService`?

Reply:
Good for me :)